.github/workflows/build_test_publish_images.yaml

```diff
@@ -79,6 +79,7 @@ jobs:
         with:
           fetch-depth: 0 # unshallow fetch for setuptools-scm
           persist-credentials: false
+          ref: ${{ inputs.sha }}
```

Comment:
I just pushed a commit adding this everywhere in the image-building and image-testing workflows.

Without this, the images might not be built from the SHA you're expecting... `actions/checkout` will by default just checkout the current ref from whatever triggered the run.

I discovered this on #359, trying to manually trigger nightlies from a different branch.

This will be most important if you ever want to run nightly builds / tests for more branches than just the current default branch:

https://github.com/NVIDIA/cuopt/blob/9006ad795e3e94e4e938a3a253ab043539154d85/.github/workflows/nightly.yaml#L12-L15